### PR TITLE
fix nav2 params and launch file to publish Local and global costmaps in multi robots example

### PR DESCRIPTION
**Pull request by Yousseflah**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #2446 |
| Primary OS tested on | (Ubuntu 20.04) |
| Robotic platform tested on | Foxy Gazebo multi-robot simulation) |

---

## Description of contribution in a few bullet points
add libgazebo_ros_init.so in launch file and change groot ports in nav2_params so the costmaps are published and navigation starts for both robots.

## Description of documentation updates required from your changes

---

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in navigation.ros.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

File: nav2_bringup/bringup/launch/multi_tb3_simulation_launch.py
Comment (SteveMacenski) [inline nav2_bringup/bringup/launch/multi_tb3_simulation_launch.py:113]: Can you make this adjustment in all of the launch files so that we don't run into this issue again when someone tries to modify it to their needs?

File: nav2_bringup/bringup/params/nav2_multirobot_params_2.yaml
Comment (SteveMacenski) [inline nav2_bringup/bringup/params/nav2_multirobot_params_2.yaml:57]: Thanks for changing the ports. Can you default these to `False` instead for the robot 1 and robot 2? That way its not running in case someone actually needs to use it

**Review by SteveMacenski — CHANGES_REQUESTED** [review]
This looks  great, I really appreciate it!

**Comment by SteveMacenski** [comment]
@Yousseflah can you submit this against the `main` branch? I'll cherry pick over to Foxy and Galactic after its merged to get CI spinning and so that we have the latest and greatest in the main master branch

**Comment by Yousseflah** [comment]
@SteveMacenski Please check.

**Review by SteveMacenski — APPROVED** [review]

**Comment by SteveMacenski** [comment]
LGTM, I'll merge tomorrow when I can also cherry pick at the same time.

**Comment by SteveMacenski** [comment]
@Yousseflah whoops, there's a linting failure
```
E501 (./launch/multi_tb3_simulation_launch.py:105:100) - nav2_bringup.flake8

line too long (109 > 99 characters):
        cmd=[simulator, '--verbose', '-s', 'libgazebo_ros_init.so', '-s', 'libgazebo_ros_factory.so', world],

```

Please make sure to run ament_flake8 and ament_pep57 (or whatever that number is, I dont recall) locally to test if your linting fixes pass to get a faster turn around on testing

**Comment by Yousseflah** [comment]
@SteveMacenski tests passed, i've fixed also a line in "nav2_gazebo_spawner.py" that wasn't related to my changes, hope this won't bother.
lints_fix

**Review by SteveMacenski — APPROVED** [review]
LGTM, waiting on CI

**Pull request merged**
